pinot-broker/src/main/java/com/linkedin/pinot/broker/requesthandler/SingleConnectionBrokerRequestHandler.java

```diff
@@ -0,0 +1,128 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.broker.requesthandler;
+
+import com.linkedin.pinot.broker.broker.AccessControlFactory;
+import com.linkedin.pinot.broker.queryquota.TableQueryQuotaManager;
+import com.linkedin.pinot.broker.routing.RoutingTable;
+import com.linkedin.pinot.broker.routing.TimeBoundaryService;
+import com.linkedin.pinot.common.config.TableNameBuilder;
+import com.linkedin.pinot.common.metrics.BrokerMeter;
+import com.linkedin.pinot.common.metrics.BrokerMetrics;
+import com.linkedin.pinot.common.metrics.BrokerQueryPhase;
+import com.linkedin.pinot.common.request.BrokerRequest;
+import com.linkedin.pinot.common.response.BrokerResponse;
+import com.linkedin.pinot.common.response.ServerInstance;
+import com.linkedin.pinot.common.response.broker.BrokerResponseNative;
+import com.linkedin.pinot.common.utils.DataTable;
+import com.linkedin.pinot.common.utils.ServerType;
+import com.linkedin.pinot.core.transport.QueryRouter;
+import com.linkedin.pinot.core.transport.ResponseFuture;
+import com.linkedin.pinot.core.transport.Server;
+import com.linkedin.pinot.core.transport.ServerResponse;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.ThreadSafe;
+import org.apache.commons.configuration.Configuration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * The <code>SingleConnectionBrokerRequestHandler</code> class is a thread-safe broker request handler using a single
+ * connection per server to route the queries.
+ */
+@ThreadSafe
+public class SingleConnectionBrokerRequestHandler extends BaseBrokerRequestHandler {
+  private static final Logger LOGGER = LoggerFactory.getLogger(SingleConnectionBrokerRequestHandler.class);
+
+  private final QueryRouter _queryRouter;
+
+  public SingleConnectionBrokerRequestHandler(@Nonnull Configuration config, @Nonnull RoutingTable routingTable,
+      @Nonnull TimeBoundaryService timeBoundaryService, @Nonnull AccessControlFactory accessControlFactory,
+      @Nonnull TableQueryQuotaManager tableQueryQuotaManager, @Nonnull BrokerMetrics brokerMetrics) {
+    super(config, routingTable, timeBoundaryService, accessControlFactory, tableQueryQuotaManager, brokerMetrics);
+    _queryRouter = new QueryRouter(_brokerId, brokerMetrics);
+  }
+
+  @Override
+  public void start() {
+  }
+
+  @Override
+  public synchronized void shutDown() {
+    _queryRouter.shutDown();
+  }
+
+  @Nonnull
+  @Override
+  protected BrokerResponse processBrokerRequest(long requestId, @Nonnull BrokerRequest originalBrokerRequest,
```

Comment:
which part of this method is specific to this implementation?

Reply:
Basically everything in this method. All the common parts have been extracted to the base class in a previous PR.